src/SerializeTool.cpp

```diff
@@ -0,0 +1,436 @@
+            { "deserialize", make_tuple(0, 1, true) },
+            { "sign", make_tuple(0, 1, true) },
+            { "multiSign", make_tuple(0, 1, true) },
+            { "create_keyfile", make_tuple(0, 2, false) },
+            { "repair_keyfile", make_tuple(0, 0, false) }
+    };
+
+    auto const iArgs = commandArgs.find(command);
+
+    if (iArgs == commandArgs.end())
+        throw std::runtime_error("Unknown command: " + command);
+
+    if (args.size() < get<0>(iArgs->second) ||
+            args.size() > get<1>(iArgs->second))
+        throw std::runtime_error("Syntax error: Wrong number of arguments");
+
+    auto const input = [&]
+    {
+        if (args.size() == 0)
+        {
+            if (get<2>(iArgs->second))
+            {
+                std::ostringstream stdinput;
+                stdinput << std::cin.rdbuf();
+                return string{ stdinput.str() };
+            }
+            else
+                return string{};
+        }
+        else
+            return args[0];
+    }();
+
+    if (command == "serialize")
+        doSerialize(input);
+    else if (command == "deserialize")
+        doDeserialize(input);
+    else if (command == "sign")
+        doSingleSign(input, keyFile);
+    else if (command == "multiSign")
+        doMultiSign(input, keyFile);
+    else if (command == "create_keyfile")
+    {
+        auto const seed = args.size() >= 2 ? args[1] : boost::optional<string>{};
+        doCreateKeyfile(keyFile, input, seed);
+    }
+    else if (command == "repair_keyfile")
+        doRepairKeyfile(keyFile);
+}
+
+//LCOV_EXCL_START
+static
+std::string
+getEnvVar (char const* name)
+{
+    std::string value;
+
+    auto const v = getenv (name);
+
+    if (v != nullptr)
+        value = v;
+
+    return value;
+}
+
+void printHelp (const boost::program_options::options_description& desc,
+    boost::filesystem::path const& defaultKeyfile)
+{
+    static std::string const name = "ripple-serialize";
+
+    std::cerr
+        << name << " [options] <command> [<argument> ...]\n"
+        << desc << std::endl <<
+R"(Commands:
+  Serialization:
+    serialize [<argument>]              Serialize from JSON.
+    deserialize [<argument>]            Deserialize to JSON.
+
+  Transaction signing:
+    sign [<argument>]                   Sign for submission.
+    multiSign [<argument>]              Apply a multi-signature.
+      Signing commands require a valid keyfile.
+      Input can be serialized or unserialized JSON.
+      Output will always be unserialized JSON.
+
+      If an <argument> is not provided, the data will be
+      read from stdin.
+
+  Key Management:
+    create_keyfile [<keytype> [<seed>]] Create a new keyfile.
+      Specifying <seed> on the command line is strongly discouraged,
+      particularly on a shared machine. Instead, create a random seed,
+      edit the keyfile "master_seed", then run repair_keyfile.
+    repair_keyfile                      Resync "master_seed"-derived fields.
```

Comment:
I feel like this could use further explanation.
You can technically put a base58-encoded seed, hex-encoded seed, or passphrase as `master_seed` when running `repair_keyfile`. I don't know how better to do it, but I think it's confusing that in order to generate keys from a hex-encoded seed, you put it in the `master_seed` field instead of `master_seed_hex`.